api/ai/priority_algorithm/mutations/random_team_size_change.py

```diff
@@ -0,0 +1,65 @@
+import random
+from typing import Dict, List
+
+from api.ai.interfaces.team_generation_options import TeamGenerationOptions
+from api.ai.priority_algorithm.custom_dataclasses import PriorityTeamSet
+from api.ai.priority_algorithm.mutations.interfaces import Mutation
+from api.ai.priority_algorithm.mutations.utils import get_available_priority_teams
+from api.ai.priority_algorithm.priority.interfaces import Priority
+from api.dataclasses.student import Student
+
+
+class RandomTeamSizeMutation(Mutation):
+    def __init__(self, num_mutations: int = 1, number_of_teams: int = 2):
+        super().__init__(num_mutations)
+        self.number_of_teams = number_of_teams
+
+    def validate(self):
+        if self.number_of_teams < 2:
+            raise ValueError("Random team size mutation must have at least two teams")
+
+    def mutate_one(
+        self,
+        priority_team_set: PriorityTeamSet,
+        priorities: List[Priority],
+        student_dict: Dict[int, Student],
+        team_generation_options: TeamGenerationOptions,
+    ) -> PriorityTeamSet:
+        if (
+            team_generation_options.max_team_size
+            - team_generation_options.min_team_size
+            < 2
+        ):
+            print(
```

Comment:
idk how valuable this is in general, if it has low effect then it's kinda like go off

Reply:
also idk if this is even necessarily true